src/mlpack/methods/ann/activation_functions/isrlu_function.cpp

```diff
@@ -0,0 +1,159 @@
+/**
+ * @file isrlu_function.hpp
+ * @author Pranav Reddy
+ *
+ * Definition and implementation of the ISRLU function
+ * mlpack is free software; you may redistribute it and/or modify it under the
+ * terms of the 3-clause BSD license.  You should have received a copy of the
+ * 3-clause BSD license along with mlpack.  If not, see
+ * http://www.opensource.org/licenses/BSD-3-Clause for more information.
+ */
+
+#ifndef MLPACK_METHODS_ANN_ACTIVATION_FUNCTIONS_ISRLU_FUNCTION_HPP
+#define MLPACK_METHODS_ANN_ACTIVATION_FUNCTIONS_ISRLU_FUNCTION_HPP
+
+#include <mlpack/prereqs.hpp>
+
+namespace mlpack{
+namespace ann /** Artifical Neural Network. */{
+
+/**
+ * The ISRU function, defined by
+ * 
+ * @f{eqnarray*}{
+ * f(x) &=& x/sqrt(1 + alpha*x*x) if x<0
+ *      &=& x                     if x>=0
+ *
+ * f'(x) &=& cube(1/sqrt(1 + alpha*x*x)) if x<0
+ *       &=& 1                           if x>=0
+ * @f
+ */
+
+class ISRLUFunction
+{
+
+ public:
+  /**
+   * Computes the ISRLU function
+   * 
+   * @param x Input data.
+   * @param alpha paramter, default value = 0.1
+   * @return f(x)
+   */
+  static double Fn(const double x, const double alpha = 0.1)
+  {
+    if (x<0)
+      return x / (std::sqrt(1 + alpha*x*x));
+    else
+      return x;
+  }
+
+  /**
+   * Computes the ISRLU function using a matrix as a input
+   * 
+   * @param x Input data.
+   * @param y The resulting output activation
+   * @param alpha parameter, default value = 0.1
+   */
+  template<typename eT>
+  static void Fn(const arma::Mat<eT>& x,
+             arma::Mat<eT>& y,
+             const double alpha = 0.1)
+  {
+    if (x<0)
+      y = x / (arma::sqrt(1 + alpha * arma::pow(x, 2)));
+    else
+      y = x;
+  }
+
+  /**
+   * Computes the ISRLU function.
+   *
+   * @param x Input data.
+   * @param y The resulting output activation.
+   * @param alpha parameter, default value = 0.1
+   */
+  template<typename InputVecType, typename OutputVecType>
+  static void Fn(const InputVecType& x,
+                 OutputVecType& y,
+                 const double alpha = 0.1)
+  {
+    y.set_size(arma::size(x));
+
+    for (size_t i = 0; i < x.n_elem; i++)
+      y(i) = Fn(x(i));
+  }
+
+  /**
+   * Computes the inverse of the ISRU function
+   * 
+   * @param y
+   * @param alpha
+   * @return f^{-1}(y)
+   */
+  static double Inv(const double y, const double alpha = 0.1)
+  {
+    if (y<0)
+      return y / std::sqrt(1 - alpha * y * y);
+    else
+      return y;
+  }
+
+  /**
+   * Computes the inverse of the ISRLU function
+   * 
+   * @param y Input data.
+   * @param x The resulting inverse of the input data
+   * @param alpha parameter, default value = 0.1
+   */
+  template<typename InputVecType, typename OutputVecType>
+  static void Inv(const InputVecType& y,
+                  OutputVecType& x,
+                  const double alpha = 0.1)
+  {
+    if (y<0)
+      x = y / arma::sqrt(1 - alpha * arma::pow(y, 2));
```

Comment:
Same as above, please prevent division by zero.

Reply:
How do you think I should handle this? Should I throw an error saying y cannot be 1/sqrt(alpha)?

Reply:
One way of handling it would be assigning DBL_MAX. I think I suggested something similar in PR of @prince776. I will take a closer look at the math for the same, Till then you should convert this implementation into layer.

Reply:
Ok I will do that thanks

Reply:
Hey @kartikdutt18 I have converted to a layer. Can you take a look at it. Also any update about the edge case?